cost/azure/savings_plan/recommendations/azure_savings_plan_recommendations.pt

```diff
@@ -0,0 +1,394 @@
+    header "User-Agent", "RS Policies"
+    ignore_status [400, 401, 404, 422]
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "name", jmes_path(col_item, "name")
+      field "subscriptionId", val(iter_item, "subscriptionId")
+      field "subscriptionName", val(iter_item, "subscriptionName")
+      field "kind", jmes_path(col_item, "kind")
+      field "totalHours", jmes_path(col_item, "properties.totalHours")
+      field "lookBackPeriod", jmes_path(col_item, "properties.lookBackPeriod")
+      field "term", jmes_path(col_item, "properties.term")
+      field "scope", jmes_path(col_item, "properties.scope")
+      field "commitmentGranularity", jmes_path(col_item, "properties.commitmentGranularity")
+      field "costWithoutBenefit", jmes_path(col_item, "properties.costWithoutBenefit")
+      field "averageUtilizationPercentage", jmes_path(col_item, "properties.recommendationDetails.averageUtilizationPercentage")
+      field "coveragePercentage", jmes_path(col_item, "properties.recommendationDetails.coveragePercentage")
+      field "commitmentAmount", jmes_path(col_item, "properties.recommendationDetails.commitmentAmount")
+      field "overageCost", jmes_path(col_item, "properties.recommendationDetails.overageCost")
+      field "benefitCost", jmes_path(col_item, "properties.recommendationDetails.benefitCost")
+      field "savingsAmount", jmes_path(col_item, "properties.recommendationDetails.savingsAmount")
+      field "savingsPercentage", jmes_path(col_item, "properties.recommendationDetails.savingsPercentage")
+      field "totalCost", jmes_path(col_item, "properties.recommendationDetails.totalCost")
+      field "wastageCost", jmes_path(col_item, "properties.recommendationDetails.wastageCost")
+    end
+  end
+end
+
+#EVALUATE DATA BASED ON SCOPE PARAMETER
+datasource "ds_sp_recommendations_data" do
+  run_script $js_eval_sp_recommendations, $ds_sp_recommendations, $param_scope
+end
+
+script "js_eval_sp_recommendations", type: "javascript" do
+  parameters "ds_sp_recommendations", "param_scope"
+  result "result"
+  code <<-EOS
+    if( param_scope == "Shared"){
+      var shared_sp_recs = []           //array of SP Recommendations excluding Subscription data
+      _.each(ds_sp_recommendations, function(rec){
+        var sp_rec = _.omit(rec, ["id", "subscriptionId", "subscriptionName"])
+        sp_rec["id"] = sp_rec.name
+        sp_rec["subscriptionId"] = "N/A"
+        sp_rec["subscriptionName"] = "All"
+        shared_sp_recs.push( sp_rec )
+      })
+      console.log(shared_sp_recs)
+      var unique_shared_sp_recs = _.uniq( shared_sp_recs, function(shared_rec){ return shared_rec.name })
+      console.log(unique_shared_sp_recs)
+      result = unique_shared_sp_recs
+    } else {
+      result = ds_sp_recommendations
+    }
+  EOS
+end
+
+#GET RECOMMENDATIONS FOR EACH RESOURCE GROUP
```

Comment:
Did we want to leave all the commented items in here?

Reply:
Removed commented out code